R/zchunk_LA144.Commercial.R

```diff
@@ -54,56 +74,453 @@ module_gcam.usa_LA144.Commercial_DISABLED <- function(command, ...) {
+      left_join(L144.in_EJ_sR9_comm_F_U_Y, by = c("subregion9", "fuel", "service", "year")) %>%
+      group_by(subregion9, fuel, service) %>%
+      mutate(value = approx_fun(year, value, rule = 2)) %>%
+      ungroup()
+
+    # At this point, we have a table of energy by subregion9, fuel, and end use that needs to be
+    # apportioned to states, and scaled
+    # The reason for apportioning to states first is that the heating and cooling energy will be modified by pop-weighted HDD and CDD
+    # prior to calculating energy shares
+    # Downscaling heating and cooling energy to states according to person-HDD and -CDD
+    CBECS_heating_fuels <- c("electricity", "gas", "refined liquids")
+    CBECS_cooling_fuels <- c("electricity", "gas")
+
+    # Expand L144.in_EJ_sR9_comm_F_U_Y heating to all states by multiplying by HDD share within subregion9
+    L144.in_EJ_state_comm_F_heating_Y <- states_subregions %>%
+      select(state, subregion9) %>%
+      # Add all fuels and years to each state
+      repeat_add_columns(tidyr::crossing(fuel = CBECS_heating_fuels, year = HISTORICAL_YEARS)) %>%
+      mutate(service = "comm heating") %>%
+      left_join_error_no_match(L144.in_EJ_sR9_comm_F_U_Y, by = c("subregion9", "fuel", "service", "year")) %>%
+      left_join_error_no_match(L143.share_state_Pop_HDD_sR9, by = c("state", "subregion9", "year")) %>%
+      mutate(value = value.x * value.y) %>%
+      select(state, subregion9, fuel, service, year, value)
+
+    # Expand L144.in_EJ_sR9_comm_F_U_Y cooling to all states by multiplying by CDD share within subregion9
+    L144.in_EJ_state_comm_F_cooling_Y <- states_subregions %>%
+      select(state, subregion9) %>%
+      repeat_add_columns(tidyr::crossing(fuel = CBECS_cooling_fuels, year = HISTORICAL_YEARS)) %>%
+      mutate(service = "comm cooling") %>%
+      left_join_error_no_match(L144.in_EJ_sR9_comm_F_U_Y, by = c("subregion9", "fuel", "service", "year")) %>%
+      left_join_error_no_match(L143.share_state_Pop_CDD_sR9, by = c("state", "subregion9", "year")) %>%
+      mutate(value = value.x * value.y) %>%
+      select(state, subregion9, fuel, service, year, value)
+
+    # Downscaling all remaining services to states according to floorspace
+    # Calculate the share of each state within its census division's (subregion9's) floorspace
+    L144.flsp_state_share_sR9 <- L144.flsp_bm2_state_comm %>%
+      group_by(subregion9, year) %>%
+      mutate(share = value / sum(value)) %>%
+      ungroup() %>%
+      select(-value)
+
+    # Remove heating and cooling
+    L144.in_EJ_sR9_comm_F_Uoth_Y <- L144.in_EJ_sR9_comm_F_U_Y %>%
+      filter(!(service %in% c("comm heating", "comm cooling")))
+
+    # Convert to state data by multiplying by floorspace share
+    L144.in_EJ_state_comm_F_Uoth_Y <- L144.in_EJ_sR9_comm_F_Uoth_Y %>%
+      select(fuel, service) %>%
+      distinct() %>%
+      repeat_add_columns(tidyr::crossing(state = gcamusa.STATES, year = HISTORICAL_YEARS)) %>%
```

Comment:
Have you gone back and put this (`gcamusa.STATES`) into previously-done gcam-usa chunks?

Reply:
Did this for chunks I had done-should I do a search through all gcam-usa chunks?

Reply:
@russellhz No, we can have original authors do that. Thanks.